samples/best_practices/disallow_host_network_hostport.yaml

```diff
@@ -1,24 +1,33 @@
 apiVersion: kyverno.io/v1alpha1
 kind: ClusterPolicy
 metadata:
-  name: validate-host-network-hostport
+  name: host-network-port
   annotations:
     policies.kyverno.io/category: Security
     policies.kyverno.io/description: Using 'hostPort' and 'hostNetwork' allows pods to share 
       the host network stack, allowing potential snooping of network traffic from an application pod.
 spec:
   rules:
-  - name: validate-host-network-hostport
+  - name: validate-host-network
     match:
       resources:
         kinds:
         - Pod
     validate:
-      message: "Defining hostNetwork and hostPort are not allowed"
+      message: "Use of hostNetwork is not allowed"
+      pattern:
+        spec:
+          =(hostNetwork): false
+  - name: validate-host-port
+    match:
+      resources:
+        kinds:
+        - Pod
+    validate:
+      message: "Use of hostPort is not allowed"
       pattern:
         spec:
-          (hostNetwork): false
           containers:
           - name: "*"
-            ports:
-            - hostPort: null
+            =(ports):
+            - hostPort: null
```

Comment:
With `hostPort:null` we expect the resource to have value null on json.
But if we want to disable the use of hostPort we should use the negation anchor `X(hostPort): <any_value>`

Reply:
I agree that the negation anchor is clearer and will update to that. 

However, both ways should work, right?

Reply:
Yes, both would work and report a violation on the same path but different reasons(with negation anchor-> field disallowed & for the null value the values won't match.